metaconfig-typesafe-config/src/main/scala/metaconfig/typesafeconfig/TypesafeConfig2Class.scala

```diff
@@ -53,36 +53,72 @@ object TypesafeConfig2Class {
               )
           }
       }
-      getPositionOpt(value.origin, input) match {
+      getPositionOpt(value.origin, cache) match {
         case Some(pos) => conf.withPos(pos)
         case None => conf
       }
     }
     try {
-      Configured.Ok(loop(config().resolve().root))
+      val resolved = config().resolve().root()
+      cache += OriginId(resolved.origin) -> input
+      Configured.Ok(loop(resolved))
     } catch {
       case e: ConfigException.Parse =>
         Configured.NotOk(
-          ConfError.parseError(getPosition(e.origin, input), e.getMessage)
+          ConfError.parseError(getPosition(e.origin, cache), e.getMessage)
         )
     }
   }
 
   private def getPosition(
       originOrNull: ConfigOrigin,
-      input: Input
+      cache: mutable.Map[OriginId, Input]
   ): Position =
-    getPositionOpt(originOrNull, input).getOrElse(Position.None)
+    getPositionOpt(originOrNull, cache).getOrElse(Position.None)
 
   private def getPositionOpt(
       originOrNull: ConfigOrigin,
-      input: Input
+      cache: mutable.Map[OriginId, Input]
   ): Option[Position] =
     for {
       origin <- Option(originOrNull)
-      linePlus1 <- Option(origin.lineNumber)
-      line = linePlus1 - 1
-      start = input.lineToOffset(line)
+      input <- cache.updateWith(OriginId(origin)) { maybeCurrent =>
+        maybeCurrent
+          .orElse(
+            Option(origin.url)
```

Comment:
this allows to short-circuit the for comprehension when we don't have a URL for the yet-unknown origin, which is the case when a key is defined across files